src/main/java/com/ust/sourcecourse/configuration/controller/DBDataSourceController.java

```diff
@@ -42,4 +44,59 @@ public ResponseEntity<List<DBDataSourceInfo>> getDBSourceInfo() {
 		return ResponseEntity.ok(dataSourceService.getDBInfo());
 	}
 
+	@DeleteMapping("/table/{id}/{tag}")
+	public ResponseEntity<String> removeTagFromSourceTable(@PathVariable("id") Long uid, @PathVariable String tag) {
+		ResponseEntity<String> updatedSourceTable = dataSourceService.removeTagFromSourceTable(uid, tag);
+		if (updatedSourceTable != null) {
+			return ResponseEntity.ok("Tag '" + tag + "' deleted successfully from SourceTable with id " + uid);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@DeleteMapping("/column/{id}/{tag}")
+	public ResponseEntity<String> removeTagFromSourceColumn(@PathVariable("id") Long uid, @PathVariable String tag) {
+		ResponseEntity<String> updatedSourceColumn = dataSourceService.removeTagFromSourceColumn(uid, tag);
+		if (updatedSourceColumn != null) {
+			return ResponseEntity.ok("Tag '" + tag + "' deleted successfully from SourceColumn with id " + uid);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@GetMapping("/table/tags/{id}")
+	public ResponseEntity<List<String>> getTagsBySourceTable(@PathVariable("id") Long uid) {
+		List<String> tags = dataSourceService.getTagsByTable(uid);
+		return ResponseEntity.ok(tags);
+	}
+
+	@GetMapping("/column/tags/{id}")
+	public ResponseEntity<List<String>> getTagsBySourceColumn(@PathVariable("id") Long uid) {
+		List<String> tags = dataSourceService.getTagsByColumn(uid);
+		return ResponseEntity.ok(tags);
+	}
+
+	@GetMapping("/table/search/{tag}")
+	public ResponseEntity<List<DBTable>> searchTablesByTag(@PathVariable String tag) {
+		List<DBTable> sourcetables = dataSourceService.searchTablesByTag(tag);
+		return ResponseEntity.ok(sourcetables);
+	}
+
+	@GetMapping("/column/search/{tag}")
+	public ResponseEntity<List<DBTableColumn>> searchColumnByTag(@PathVariable String tag) {
+		List<DBTableColumn> groups = dataSourceService.searchcolumnByTag(tag);
+		return ResponseEntity.ok(groups);
+	}
+	@PostMapping("table/post/{tags}")
+    public ResponseEntity<List<String>> addTagToSourceTable(@PathVariable ("id") Long uid, @RequestBody  List<String> tags) {
+        List<String> updatedSourceTable= dataSourceService.addTagSourceTable(uid, tags);
+        return ResponseEntity.ok(updatedSourceTable);
+    }
+	@PostMapping("column/post/{tags}")
+    public ResponseEntity<List<String>> addTagToSourceColumn(@PathVariable ("id") Long uid, @RequestBody  List<String> tags) {
```

Comment:
This should be accepting TagDescriptionRequest, not just tags alone. Updating description also should be possible

Reply:
No need of POST in URL

Reply:
resolved